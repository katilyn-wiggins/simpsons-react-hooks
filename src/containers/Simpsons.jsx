import React, { useState } from 'react'; 
import Load from '../components/quote/Load'; 
import Quote from '../components/quote/Quote';
import { getQuote } from '../services/simpsonsApi'; 

const Simpsons = () => {
    const [quote, setQuote] = useState({}); 


    const handleClick = async() => {
        const quote = await getQuote(); 
        setQuote(quote);
    }; 
    
    
    return(
        <>
            <h1>Simpsons Quote Generator</h1>
            <Load onClick={handleClick} /> 
            <Quote quote={quote} /> 
        </>
    ); 
} 


export default Simpsons; 