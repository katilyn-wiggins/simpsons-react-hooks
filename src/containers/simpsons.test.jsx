import React from 'react';
import { fireEvent, render, screen, waitFor } from '@testing-library/react'; 
import Simpsons from './Simpsons'; 


describe('Simpsons Container', () => {
    it('returns a quote on button click', async() => {
        render(<Simpsons />); 

        const button = await screen.getByRole('button'); 
        fireEvent.click(button); 

        return waitFor(() => {

        });
    });
});