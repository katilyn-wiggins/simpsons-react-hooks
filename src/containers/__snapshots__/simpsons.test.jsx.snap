// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simpsons Container returns a quote on button click 1`] = `
<figure
  aria-label="character-quote"
  role="quote"
>
  <img />
  <p />
  <p />
</figure>
`;
